### PR TITLE
fix: max write buffering

### PR DESCRIPTION
**Pull request by ronag**
The current logic looks wrong. I don't know exactly what kind of behaviour it would cause but I believe this version is more correct.

### CODE REVIEW

**Comment by ronag** [comment]
I'm not sure what the failing tests is testing. Doesn't quite make sense to me? Maybe someone could clarify?

**Comment by mcollina** [comment]
> I'm not sure what the failing test is testing. Doesn't quite make sense to me? Maybe someone could clarify?

what test? seems CI is passing.

**Comment by ronag** [comment]
> > I'm not sure what the failing test is testing. Doesn't quite make sense to me? Maybe someone could clarify?
> 
> what test? seems CI is passing.

My mistake. Missed resetting a file when testing.

File: test.js [hunk 1]
```diff
@@ -1329,40 +1329,40 @@ for (const fd of [1, 2]) {
...
-    }
-
-    stream.end()
-  })
-
-  stream.on('close', () => {
-    t.pass('close emitted')
-  })
-})
+// test('write enormously large buffers async atomicly', (t) => {
```
Comment (mcollina) [inline test.js:1332]: Why was this commented out?

**Review by mcollina — COMMENTED** [review]
Could you add a test for this?

File: test.js (hunk 1 above)
Comment (ronag) [reply test.js:1332]: Aah! That was the test that didn't work. I don't understand it. What is it testing?

Comment (mcollina) [reply test.js:1332]: That one element inside the `._bufs` array is written atomicly.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**